### PR TITLE
refactor: improve frontend polling

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
-   [x] https://app.clickup.com/t/865cz1vzc
-   [x] https://app.clickup.com/t/866ajg8y7
-   [x] https://app.clickup.com/t/862k8tz1h

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: resources/js/Hooks/useWalletPollingIntervals.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ) {
+        return Interval.Short;
+    }
+
+    if (wallet?.timestamps.tokens_fetched_at !== null && wallet?.timestamps.native_balances_fetched_at !== null) {
+        return Interval.Long;
+    }
+
+    if (isTruthy(retries) && retries > 12) {
+        return Interval.Regular;
```
Comment (shahin-hq) [inline resources/js/Hooks/useWalletPollingIntervals.ts:34]: A question for you @goga-m:
If I understood after a transaction we mark the timestamp and use it to decide at what intervals we should pull data from backend. Don't you think that we should do it after a tx minted?

File: resources/js/Components/PortfolioBreakdown/Hooks/usePortfolioBreakdown.ts [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
-        );
-
-        setAssets(data);
-
-        setLoading(false);
-    };
+    const { data, isLoading } = useQuery({
+        enabled: isTruthy(wallet),
+        queryKey,
+        refetchInterval: () => calculateInterval(queryClient.getQueryState(queryKey)?.dataUpdateCount),
```
Comment (oXtxNt9U) [inline resources/js/Components/PortfolioBreakdown/Hooks/usePortfolioBreakdown.ts:21]: Am I missing something or are we potentially passing `null/undefined` for `CalculateIntervalProperties`? 

The signature here https://github.com/ArdentHQ/dashbrd/pull/24/files/13a52a45da97af5e329b7292f2178bdb89c40a30#diff-31a1e71a0d82b031fc5bbf8ff7bb1ac29de33f4dc2ccfbd7a9b5896fa7792fefR21 should be changed to reflect it.

Also double check if `dataUpdateCount` is actually an object.

**Review by oXtxNt9U — APPROVED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Comment by ItsANameToo** [comment]
seems to work with what I tested, but would like to have @samharperpittam on this before merging

**Comment by ItsANameToo** [comment]
@samharperpittam for easier testing, adjust the interval in https://github.com/ArdentHQ/dashbrd/pull/24/files#diff-31a1e71a0d82b031fc5bbf8ff7bb1ac29de33f4dc2ccfbd7a9b5896fa7792fefR6 to be something like 10 seconds instead of 5 minutes

**Comment by samharperpittam** [comment]
@goga-m @ItsANameToo  - I changed the interval to 10 seconds as recommended for easier testing. Sent a transaction from one of my wallets from Metamask (not using Dashbrd's send functionality) and it's been 7 minutes now and my balance hasn't updated. Can you confirm if you have the same issue? 

Using the send function within dashbrd updates balances very quickly.

**Review by samharperpittam — APPROVED** [review]
Ignore previous comment, user error

**Pull request merged**